lms/djangoapps/lti_provider/tests/test_views.py

```diff
@@ -0,0 +1,180 @@
+
+from lti_provider import views
+from lti_provider.signature_validator import SignatureValidator
+from student.tests.factories import UserFactory
+
+
+def set_lti_parameters(dictionary):
+    """
+    Helper method to set up a valid LTI launch.
+    """
+    dictionary['roles'] = u'Instructor,urn:lti:instrole:ims/lis/Administrator'
+    dictionary['context_id'] = u'lti_launch_context_id'
+    dictionary['oauth_version'] = u'1.0'
+    dictionary['oauth_consumer_key'] = u'consumer_key'
+    dictionary['oauth_signature'] = u'OAuth Signature'
+    dictionary['oauth_signature_method'] = u'HMAC-SHA1'
+    dictionary['oauth_timestamp'] = u'OAuth Timestamp'
+    dictionary['oauth_nonce'] = u'OAuth Nonce'
+
+
+class LtiLaunchTest(TestCase):
+    """
+    Tests for the lti_launch view
+    """
+
+    def setUp(self):
+        super(TestCase, self).setUp()
+        settings.FEATURES['ENABLE_LTI_PROVIDER'] = True
+        # Always accept the OAuth signature
+        SignatureValidator.verify = MagicMock(return_value=True)
+
+    def build_request(self, authenticated=True):
+        """
+        Helper method to create a new request object for the LTI launch.
+        """
+        request = RequestFactory().post('/')
+        request.user = UserFactory.create()
+        request.user.is_authenticated = MagicMock(return_value=authenticated)
+        request.session = {}
+        set_lti_parameters(request.POST)
+        return request
+
+    def test_valid_launch(self):
+        """
+        Verifies that the LTI launch succeeds when passed a valid request.
+        """
+        request = self.build_request()
+        response = views.lti_launch(request, None, None)
+        self.assertEqual(response.status_code, 200)
+
+    def launch_with_missing_parameter(self, missing_param):
+        """
+        Helper method to remove a parameter from the LTI launch and call the view
+        """
+        request = self.build_request()
+        del request.POST[missing_param]
+        return views.lti_launch(request, None, None)
+
+    def test_launch_with_missing_parameters(self):
+        """
+        Runs through all required LTI parameters and verifies that the lti_launch
+        view returns Bad Request if any of them are missing.
+        """
+        for missing_param in views.REQUIRED_PARAMETERS:
+            response = self.launch_with_missing_parameter(missing_param)
+            self.assertEqual(
+                response.status_code, 400,
+                'Launch should fail when parameter ' + missing_param + ' is missing')
+
+    def test_launch_with_disabled_feature_flag(self):
+        """
+        Verifies that the LTI launch will fail if the ENABLE_LTI_PROVIDER flag
+        is not set
+        """
+        settings.FEATURES['ENABLE_LTI_PROVIDER'] = False
+        request = self.build_request()
+        response = views.lti_launch(request, None, None)
+        self.assertEqual(response.status_code, 403)
+
+    # pylint: disable=unused-argument
```

Comment:
I like that you're using the error name rather than number. However, I prefer to have the disables scoped as narrowly as possible - so that is inline with the violation:

```python
def fun(unused):  # pylint: disable=unused-argument
```

Or, for unused arguments, you can indicate you know they're unused by prefixing a `_` to the variable name. This I very much prefer, because if you end up having multiple arguments on a line, it's clear which one or ones are unused, and which ones are used.